repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -197,18 +230,62 @@ object Evaluator{
       // Exhaust the printer iterator now, before exiting the `Catching`
       // block, so any exceptions thrown get properly caught and handled
       evaluatorRunPrinter(printer(evalMain(cls).asInstanceOf[Iterator[String]]))
+      evaluationResult(wrapperName, newImports)
+    }
+
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
```

Comment:
That makes sense, but doesn't `compileCacheBlock` only get used in one place where we're going to immediately evaluate it anyway? At least that's the only call site I see in this diff. I don't see any reason that normal command-line input couldn't do the same caching that loading scripts does, and it seems to me that it could simplify this code considerably